### DISCUSSION TITLE
Extension for default API's?

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#14291] [Archive]
Hi!

We have a requirement where we should return a field "price" in the "product" collection, where the value is a calculation of "list_price" in the "product" collection, subtracted by the "discount" field from the "discounts" collection.

Is there any way to do this, without storing the calculated prices OR creating a separate custom endpoint?

Cheers!

Kees.

**Comment by br41nslug** [comment]
Hi,
There is unfortunately no way to do that at the moment without using either a custom endpoint or hooks/flows to persist the calculated prices that i can think of.

**Comment by rijkvanzanten** [comment]
You could theoretically use the `read` filter hook, though that might get pretty heavy if you do it for every item that's read

**Reply by keesvanbemmel** [reply]
I'll try to see what that does performance wise... creative @rijkvanzanten
